### PR TITLE
[WIP] tables

### PR DESCRIPTION
**Pull request by Mandily**
I've started some work to clean up the tables in the Solidus admin. I think a lot of the existing table code is redundant now that we've got Bootstrap in the project, so I removed what I could to clean it up, and then added a couple of styles back in to "clean up" what was left. I've just tackled the index pages (except for the stock page) at the moment, and I'd like some feedback before I go through the rest of the views. 

Orders
screen shot 2017-07-13 at 5 01 01 pm

Products
screen shot 2017-07-13 at 5 01 19 pm

Reports
screen shot 2017-07-13 at 5 00 41 pm

Promotions
screen shot 2017-07-13 at 5 05 50 pm

Users
screen shot 2017-07-13 at 5 22 22 pm

@Sinetheta  @jhawthorn @tvdeyen @graygilmore can I get your thoughts on this?

### CODE REVIEW

**Comment by cbrunsdon** [comment]
I'm hardly the right person to comment, but everything here looks reasonable to me.

Though we'll likely have to do a semi-large manual QA to make sure there aren't tables somewhere in the deep dark parts of solidus we will need to change.

**Comment by tvdeyen** [comment]
I will make a QA session tomorrow morning

**Comment by graygilmore** [comment]
Cool! Going to run through this now.

**Comment by graygilmore** [comment]
The stock management table in particular falls apart here. I had already been working on some changes to that table. I will coordinate with @Mandily to clean up that area.

**Comment by tvdeyen** [comment]
Nice work @Mandily 👏  

**Sorry in advance** - this is hence the nature of this change - a very long review comment 😞

**I _really_ like the new table style**. Hence tables are a mayor UI pattern in Solidus admin there is still some work to do. 

Walking through the admin I found these issues:

1. All what @graygilmore said

2. `align-center` should be **removed from nearly** all tables (some examples below) or the header also needs to be centered (like in the "Countries" table -> "States required" column)

countries - locations - settings 2017-07-18 09-42-44

tax categories - taxes - settings 2017-07-18 09-44-01

zones - locations - settings 2017-07-18 09-30-59

payment methods - payments - settings 2017-07-18 09-30-28

3. Some tables has **hard coded inline styles** for `padding-left` or `margin-left` (like the "Product Properties" and "Reports" table)

reports

monosnap 2017-07-18 09-26-50

4. Some tables with a "sorting handle" miss an empty header column - otherwise the second column header looks misplaced

product properties - product 10 - 8963 - products 2017-07-18 09-22-35

product properties - product 10 - 8963 - products 2017-07-18 09-22-24

monosnap 2017-07-18 09-24-26

5. We should consider to **top align all table cells**. Tables that have more than one row though word wrapping look weird (like the "Order / Payments" or the "User / Store Credit" table)

order payments

store credit - mikki_langworth satterfield co uk - users 2017-07-18 09-55-09

Again, great work. Please feel free to contact me for help. Maybe we should even consider to split this PR up into smaller PRs.

Thanks for the great work!

**Comment by graygilmore** [comment]
@tvdeyen beauty! I nabbed the `align-center`s yesterday but I'll grab these other ones, too. Just need access to push to @Mandily's PR. I also have a commit that removes the `cycle()` for odd/even classes.

There might also be some `no-border` classes that can go away now, too. We'll have clean tables in no time!

**Comment by graygilmore** [comment]
I believe I've nabbed all of @tvdeyen's concerns as well as a couple of my own. This PR is rebased and I believe ready for a proper review.

**Comment by tvdeyen** [comment]
@graygilmore @Mandily https://github.com/Mandily/solidus/pull/1

File: backend/app/assets/stylesheets/spree/backend/shared/_tables.scss [hunk 1]
```diff
@@ -4,27 +4,9 @@ table {
   border-spacing: 0;
 
   th, td {
-    border-right: 1px solid $color-border;
-    border-bottom: 1px solid $color-border;
-    vertical-align: middle;
+    vertical-align: top;
```
Comment (Mandily) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: I actually prefer the before to the after. As a general rule, I try to keep less space between the lines of the same label/list item/paragraph and more between it and whatever is around. It helps keep the item grouped with itself a little better.

Comment (graygilmore) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: I'm with @Mandily on this one, not a fan of the large line-height.

Comment (tvdeyen) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: I agree that the large line-height is not ideal, but I also think that the vertical alignment of single line rows should be corrected. What makes me thinking is that the default tables in Bootstraps documentation are all correctly aligned. 🤔

I will have a second look into this problem

Comment (tvdeyen) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: @graygilmore @Mandily I know why the vertical alignment for single line rows is off. The action buttons occupy more space then the text and thats why the row is too high. 

monosnap 2017-07-19 21-09-55

Solution would be to get rid of the round background. 

option types - products 2017-07-19 21-13-06

We should handle this problem in separate PR. 
WDYT about this approach?

Comment (tvdeyen) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: https://github.com/solidusio/solidus/pull/2100

Comment (graygilmore) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: Separate PR makes sense. I think we're due for another solution in that area entirely.

Comment (Mandily) [inline backend/app/assets/stylesheets/spree/backend/shared/_tables.scss]: A bunch of these (this particular example included) can be removed and replaced with the inline editing solution that Martin worked on. I linked a drive document in #2042 for those. I'm ok with this style for the ones that remain.

**Comment by graygilmore** [comment]
Is there any other work required to get this into an acceptable state?

**Comment by tvdeyen** [comment]
- [ ] We redundantly declare `vertical-align: top` on `table td` although this is Bootstrap's default
- [ ] Sortable tables should have vertically centered table rows
- [ ] The reimbursements table of customer return is horizontally centered. It sits in a `fieldset` with `align-center` class.

All of the above are solved in https://github.com/Mandily/solidus/pull/2

**Comment by tvdeyen** [comment]
@graygilmore FYI

**Comment by jhawthorn** [comment]
Merged as #2092

**Pull request closed without merging**